packages/plugins/gatsby-preview/admin/src/components/SettingsList/index.js

```diff
@@ -0,0 +1,152 @@
+import styled from 'styled-components';
+import { Box } from '@strapi/design-system/Box';
+import { Tab, TabGroup, TabPanel, TabPanels, Tabs } from '@strapi/design-system/Tabs';
+import { useIntl } from 'react-intl';
+import { useNotification } from '@strapi/helper-plugin';
+import { useMutation, useQueryClient } from 'react-query';
+import axiosInstance from '../../utils/axiosInstance';
+import getTrad from '../../utils/getTrad';
+import pluginId from '../../pluginId';
+import ContentTypes from '../ContentTypes';
+import SyncSettings from '../SyncSettings';
+import TAB_LABELS from './utils/tabLabels';
+
+const StyledBox = styled(Box)`
+  overflow-x: auto;
+`;
+
+const sortContentTypes = contentTypes => sortBy(contentTypes, ct => ct.info.singularName);
+
+const SettingsList = forwardRef(
+  ({ contentTypes, formErrors, initialData, onToggleDisplaySaveButton }, ref) => {
+    const { formatMessage } = useIntl();
+    const toggleNotification = useNotification();
+    const queryClient = useQueryClient();
+    const [modifiedData, setModifiedData] = useState(initialData.previews);
+    const [contentSyncURL, setContentSyncURL] = useState(initialData.contentSyncURL);
+
+    // Because of the tabs and since we only display the save button when the settings tab is activated we need this trick to retrieve the
+    // content sync url
+    // Plus it avoids having to synchronize the state with the useQueries hook
+    useImperativeHandle(ref, () => ({
+      getContentSyncURL: () => {
+        return { contentSyncURL };
+      },
+    }));
+
+    const mutation = useMutation(
+      body => {
+        return axiosInstance.put(`/${pluginId}/previews`, body);
+      },
+      {
+        // Opitmistic response
+        onMutate: async body => {
+          await queryClient.cancelQueries(`${pluginId}-settings`);
+
+          const previousResponse = queryClient.getQueryData(`${pluginId}-settings`);
+
+          queryClient.setQueryData(`${pluginId}-settings`, old => ({ ...old, previews: body }));
+
+          return { previousResponse };
+        },
+        onSuccess: () => {
+          toggleNotification({
+            type: 'success',
+            message: {
+              id: getTrad('Settings.save-sucess'),
+              defaultMessage: 'Settings have been updated',
+            },
+          });
+        },
+      }
+    );
+
+    const collectionTypes = sortContentTypes(
+      contentTypes.filter(({ kind }) => kind === 'collectionType')
+    );
+    const singleTypes = sortContentTypes(contentTypes.filter(({ kind }) => kind === 'singleType'));
+
+    const handleChange = uid => {
+      const updatedState = { ...modifiedData, [uid]: !modifiedData[uid] };
+      setModifiedData(updatedState);
+
+      mutation.mutate(updatedState);
+    };
+
+    const handleChangeContentSyncURL = ({ target: { value } }) => {
+      setContentSyncURL(value);
+    };
+
+    const handleTabChange = tabIndex => onToggleDisplaySaveButton(tabIndex === 2);
```

Comment:
Could it be interesting if each tab gets an id, so it would be possible to link to one pane? In case the target is passed through the URL, we could then open the corresponding pane, ex: http://localhost:4000/admin/settings/gatsby-preview#single-types

In case you like the idea, the hash could be appended to the URL here ..

Reply:
I not sure it's possible with the DS component.